docs/particles.md

```diff
@@ -15,19 +15,27 @@ default contain `x`, `y`, and `z` `ParticleVariable`s; additional fields can be
 ```c++
 Swarm.Add(name, metadata)
 ```
-Each `Swarm` belongs to a `MeshBlock`, which is pointed to by `Swarm::pmy_block`.
+For a given species, each `MeshBlock` contains its own `Swarm` that holds the particles of
+that species that are spatially contained by that `MeshBlock`. The `MeshBlock` is pointed
+to by `Swarm::pmy_block`.
+
+The `Swarm` is a host-side object, but some of its data members are required for device-
+side compution. To access this data, a `SwarmDeviceContext` object is created via
+`Swarm::GetDeviceContext()`. This object can then be passed by copy into Kokkos lambdas.
+Hereafter we refer to it as `swarm_d`.
 
 To add particles to a `Swarm`, one calls
 ```c++
-ParArrayND<bool> new_particles_mask = swarm->AddEmptyParticles(num_to_add)
+ParArrayND<bool> new_particles_mask = swarm->AddEmptyParticles(num_to_add, new_indices)
 ```
 This call automatically resizes the memory pools as necessary and returns a
 `ParArrayND<bool>` mask indicating which indices in the `ParticleVariable`s are newly
-available.
+available. `new_indices` is a reference to a `ParArrayND<int>` of size `num_to_add` which
+contains
```

Comment:
I think there's sth missing at the end of this sentence, isn't it?

Reply:
Yes thanks, not sure what happened there.